tensorflow_model_optimization/g3doc/guide/clustering/index.md

```diff
@@ -0,0 +1,99 @@
+# Weight clustering
```

Comment:
Good question. It appears that the original paper uses both 'weight sharing' and 'clustering' (on its own), while arguably refering to 'clustering' in a slightly narrower context of initial clustering of weights before the fine tuning. It seems to me the boundary is quite vague but it would make sense to refine the use of these terms.
@arovir01, how about changing the title of this section to 'Clustering', and introduce the interchangeability of both terms in this paragraph, instead of the beginning of the 'Overview' section below? E.g.
"overview on weight clustering to help" -> overview on clustering, or weight sharing, to help"

Reply:
@akarmi Yes, I can do that.

Reply:
I believe the "Overview" title should be kept to stay in line with other similar documents, such as https://github.com/tensorflow/model-optimization/blob/master/tensorflow_model_optimization/g3doc/guide/quantization/training.md